pkg/kubelet/cri/remote/utils_test.go

```diff
@@ -0,0 +1,195 @@
+		expectedResult:   fmt.Errorf("status.CreatedAt is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of name",
+		podSandboxStatus: getPodSandboxStatus(true, true, true, false),
+		expectedResult:   fmt.Errorf("metadata.Name, metadata.Namespace or metadata.Uid is not in metadata \"&PodSandboxMetadata{Name:,Uid:100,Namespace:foo-ns,Attempt:10,}\""),
+	}}
+
+	for _, tc := range testcases {
+		t.Run(tc.description, func(t *testing.T) {
+			result := verifySandboxStatus(tc.podSandboxStatus)
+			assert.Equal(t, tc.expectedResult, result)
+		})
+	}
+}
+
+func getPodSandboxStatus(validId, validMetadata, validCreatedAt, validName bool) *runtimeapi.PodSandboxStatus {
+	id := "12345678"
+	metadata := &runtimeapi.PodSandboxMetadata{
+		Name:          "testname",
+		Uid:           "100",
+		Namespace:     "foo-ns",
+		Attempt:       10,
+		XXX_sizecache: 100,
+	}
+	createAt := int64(6000)
+
+	if !validId {
+		id = ""
+	}
+	if !validMetadata {
+		metadata = nil
+	}
+	if !validCreatedAt {
+		createAt = 0
+	}
+	if !validName {
+		metadata.Name = ""
+	}
+	return &runtimeapi.PodSandboxStatus{
+		Id:        id,
+		Metadata:  metadata,
+		State:     runtimeapi.PodSandboxState_SANDBOX_READY,
+		CreatedAt: createAt,
+		Network: &runtimeapi.PodSandboxNetworkStatus{
+			Ip:            "192.168.1.100",
+			XXX_sizecache: 100,
+		},
+		Linux: &runtimeapi.LinuxPodSandboxStatus{
+			XXX_sizecache: 100,
+		},
+		RuntimeHandler: "foo-header",
+		XXX_sizecache:  100,
+	}
+}
+
+func TestVerifyContainerStatus(t *testing.T) {
+
+	testcases := []struct {
+		description     string
+		containerStatus *runtimeapi.ContainerStatus
+		expectedResult  error
+	}{{
+		description:     "verifyContainerStatus succeeds",
+		containerStatus: getContainerStatus(true, true, true, true, true),
+		expectedResult:  nil,
+	}, {
+		description:     "verifyContainerStatus fails due to lack of Id",
+		containerStatus: getContainerStatus(false, true, true, true, true),
+		expectedResult:  fmt.Errorf("status.Id is not set"),
+	}, {
+		description:     "verifyContainerStatus fails due to lack of metadata",
+		containerStatus: getContainerStatus(true, false, true, true, true),
+		expectedResult:  fmt.Errorf("status.Metadata is not set"),
+	}, {
+		description:     "verifyContainerStatus fails due to lack of CreatedAt",
+		containerStatus: getContainerStatus(true, true, false, true, true),
+		expectedResult:  fmt.Errorf("status.CreatedAt is not set"),
+	}, {
+		description:     "verifyContainerStatus fails due to lack of ImageRef",
+		containerStatus: getContainerStatus(true, true, true, false, true),
+		expectedResult:  fmt.Errorf("status.ImageRef is not set"),
+	}, {
+		description:     "verifyContainerStatus fails due to lack of name",
+		containerStatus: getContainerStatus(true, true, true, true, false),
+		expectedResult:  fmt.Errorf("metadata.Name is not in metadata \"&ContainerMetadata{Name:,Attempt:10,}\""),
+	}}
```

Comment:
verifyContainerStatus returns an error when Status.Image is nil.
https://github.com/kubernetes/kubernetes/blob/5309df33a9e4f34f0124bceb97c91c205999d546/pkg/kubelet/cri/remote/utils.go#L82-L84
Is there any reason why not to include this in the test case?

Reply:
A nice catch.
OK, I will try adding the test case also.

Reply:
Thanks for your review, done.

Reply:
Thanks for adding the test case.
It looks good to me but unfortunately there are conflicts.
Please rebase.

Reply:
Thanks for letting me know that.
I will do that soon.

Reply:
/lgtm